EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockGEM.cc

```diff
@@ -0,0 +1,257 @@
+        // int _sector_gem = (_subsector < 5) ? _sector : (_sector % 6) + 1; //
+        int _sector_gem = _sector;
+        // Rotate by 2 to match GEM convention in CMSSW (GEMDetId.h) // FIXME VERIFY
+        int _subsector_gem = ((_subsector + 1) % 6) + 1;
+        // Define chamber number) // FIXME VERIFY
+        int _chamber = (_sector_gem - 1) * 6 + _subsector_gem;
+        // Define CSC-like subsector) // FIXME WHY?? VERIFY
+        int _subsector_csc = (_station != 1) ? 0 : ((_chamber % 6 > 2) ? 1 : 2);
+
+        Hit_.set_station(_station);
+        Hit_.set_ring(_ring);
+        Hit_.set_sector(_sector);
+        Hit_.set_subsector(_subsector_csc);
+        Hit_.set_chamber(_chamber);
+        Hit_.set_neighbor(_neighbor);
+
+        // Fill the EMTFHit
+        ImportGEM(Hit_, GEM_, (res->at(iOut)).PtrEventHeader()->Endcap(), (res->at(iOut)).PtrEventHeader()->Sector());
+
+        // Set the stub number for this hit
+        // Each chamber can send up to 2 stubs per BX // FIXME is this true for GEM, are stubs relevant for GEMs?
+        // Also count stubs in corresponding CSC chamber; GEM hit counting is on top of LCT counting
+        Hit_.set_stub_num(0);
+        // See if matching hit is already in event record
+        bool exact_duplicate = false;
+        for (auto const& iHit : *res_hit) {
+          if (Hit_.BX() == iHit.BX() && Hit_.Endcap() == iHit.Endcap() && Hit_.Station() == iHit.Station() &&
+              Hit_.Chamber() == iHit.Chamber()) {
+            if ((iHit.Is_CSC() == 1 && iHit.Ring() == 2) ||
+                (iHit.Is_GEM() == 1)) {  // Copied from RPC, but GEM has no ring 2/3...
+              if (Hit_.Neighbor() == iHit.Neighbor()) {
+                Hit_.set_stub_num(Hit_.Stub_num() + 1);
+                if (iHit.Is_GEM() == 1 && iHit.Ring() == Hit_.Ring() && iHit.Roll() == Hit_.Roll() &&
+                    iHit.Pad() == Hit_.Pad()) {
+                  exact_duplicate = true;
+                }
+              }
+            }
+          }
+        }  // End loop: for (auto const & iHit : *res_hit)
+
+        if (exact_duplicate)
+          edm::LogWarning("L1T|EMTF") << "EMTF unpacked duplicate GEM digis: BX " << Hit_.BX() << ", endcap "
+                                      << Hit_.Endcap() << ", station " << Hit_.Station() << ", neighbor "
+                                      << Hit_.Neighbor() << ", ring " << Hit_.Ring() << ", chamber " << Hit_.Chamber()
+                                      << ", roll " << Hit_.Roll() << ", pad " << Hit_.Pad() << std::endl;
+
+        (res->at(iOut)).push_GEM(GEM_);
+        if (!exact_duplicate)
+          res_hit->push_back(Hit_);
+
+        if (!exact_duplicate)
+          res_GEM->insertDigi(Hit_.GEM_DetId(), Hit_.CreateGEMPadDigiCluster());
+
+        // Finished with unpacking one GEM Data Record
+        return true;
+
+      }  // End bool GEMBlockUnpacker::unpack
+
+      // bool GEMBlockPacker::pack(const Block& block, UnpackerCollections *coll) {
```

Comment:
please remove commented-out code.

Reply:
Commented code reflects state of other EMTF block unpackers, and represents placeholders for the needed and future packer code